ime/app/src/main/java/com/anysoftkeyboard/gesturetyping/GestureTypingDetector.java

```diff
@@ -26,6 +26,19 @@ public class GestureTypingDetector {
   private static final int CURVATURE_NEIGHBORHOOD = 1;
   private static final double MINIMUM_DISTANCE_FILTER = 1000000;
 
+  /**
+   * Maximum squared distance from gesture start point to accept a word's starting key.
+   * This allows for imprecise gesture starts while filtering obviously wrong candidates.
+   * Value is approximately 1.5 key widths squared (assuming ~100 pixel keys).
+   */
+  private static final int START_KEY_PROXIMITY_THRESHOLD_SQUARED = 22500;
```

Comment:
This should be a resource. Probably add that to ime/app/src/main/res/values/dimens.xml. For example, instead of 100 pixels, you may add `<dimen name="gesture_typing_proximity_threshold">20dp</dimen>` (or whatever value works for you). Then you can use `context.getResources().getDimensionPixelSize(R.dimen.gesture_typing_proximity_threshold)`

Or something like that.

Reply:
150 dp depends on the device. Mine will be 450 pixels.
<img width="524" height="316" alt="Screenshot 2025-12-15 at 3 37 17 PM" src="https://github.com/user-attachments/assets/69d1fe38-2a14-4d52-ba36-2dc8d7884265" />

Consider experimenting with different values to see how performance and accuracy changes.